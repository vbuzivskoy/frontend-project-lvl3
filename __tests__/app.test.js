import fs from 'fs';
import path from 'path';
import nock from 'nock';
import axios from 'axios';
import { waitFor, getByPlaceholderText, getByText } from '@testing-library/dom';
import { toBeInTheDocument, toBeEnabled } from '@testing-library/jest-dom/matchers';
import userEvent from '@testing-library/user-event';
import prettier from 'prettier';
import app from '../src/app';

debugger; // eslint-disable-line no-debugger

expect.extend({ toBeInTheDocument, toBeEnabled });

axios.defaults.adapter = require('axios/lib/adapters/http');

nock.disableNetConnect();

const getFixturePath = (filename) => path.join(__dirname, '__fixtures__', filename);
const getSourcePath = (filename) => path.join(__dirname, '..', 'src', filename);
const readFixture = (filename) => fs.readFileSync(getFixturePath(filename), 'utf-8');
const copyFixture = (filename) => (
  fs.copyFileSync(getSourcePath(filename), getFixturePath(filename))
);
const removeFixture = (filename) => (
  fs.unlinkSync(getFixturePath(filename))
);

const prettierOptions = {
  parser: 'html',
  htmlWhitespaceSensitivity: 'ignore',
  tabWidth: 4,
};
const getTree = () => prettier.format(document.body.innerHTML, prettierOptions);

const proxyHost = 'https://cors-anywhere.herokuapp.com';
const wrongUrl = 'htp://lorem-rss.herokuapp.com/feed';
const rssFeedUrl1sec = 'http://lorem-rss.herokuapp.com/feed?length=5&unit=second&interval=1';
const rssFeedUrl2sec = 'http://lorem-rss.herokuapp.com/feed?length=5&unit=second&interval=2';
const rssXml1sec1 = readFixture('rss1sec1.xml');
const rssXml2sec1 = readFixture('rss2sec1.xml');
const rssXml2sec2 = readFixture('rss2sec2.xml');
let elements;

beforeAll(() => copyFixture('index.html'));
afterAll(() => removeFixture('index.html'));

beforeEach(() => {
  const initHtml = readFixture('index.html');
  document.documentElement.innerHTML = initHtml;
  return app()
    .then(() => {
      elements = {
        input: getByPlaceholderText(document.body, 'RSS link'),
        button: getByText(document.body, 'Add'),
      };
    });
});

test('Wrong url test', () => {
  userEvent.paste(elements.input, wrongUrl);
  return waitFor(() => {
    const validationFeedback = getByText(document.body, 'The field should contains valid URL');
    expect(validationFeedback).toBeInTheDocument();
  });
});

test('Empty url test', () => {
  userEvent.paste(elements.input, wrongUrl);
  return waitFor(() => {
    const validationFeedback = getByText(document.body, 'The field should contains valid URL');
    expect(validationFeedback).toBeInTheDocument();
    userEvent.clear(elements.input);
  })
    .then(() => (
      waitFor(() => {
        const validationFeedback = getByText(document.body, 'The field should not be empty');
        expect(validationFeedback).toBeInTheDocument();
      })
    ));
});

test('One rss feed test', () => {
  expect(getTree()).toMatchSnapshot();
  nock(proxyHost)
    .get(`/${rssFeedUrl2sec}`)
    .twice()
    .reply(200, rssXml2sec1);

  userEvent.paste(elements.input, rssFeedUrl2sec);
  return waitFor(() => {
    expect(elements.button).toBeEnabled();
  })
    .then(() => {
      userEvent.click(elements.button);
    })
    .then(() => waitFor(() => {
      const text = getByText(document.body, 'Lorem ipsum 2020-07-02T10:46:16Z');
      expect(text).toBeInTheDocument();
      expect(getTree()).toMatchSnapshot();
    }))
    .then(() => {
      nock(proxyHost)
        .get(`/${rssFeedUrl2sec}`)
        .once()
        .reply(200, rssXml2sec2);
    })
    .then(() => waitFor(() => {
      const text = getByText(document.body, 'Lorem ipsum 2020-07-02T10:46:20Z');
      expect(text).toBeInTheDocument();
      expect(getTree()).toMatchSnapshot();
    }, { timeout: 6000 }));
}, 10000);

test('Two rss feed test', () => {
  expect(getTree()).toMatchSnapshot();
  nock(proxyHost)
    .get(`/${rssFeedUrl2sec}`)
    .twice()
    .reply(200, rssXml2sec1);

  userEvent.paste(elements.input, rssFeedUrl2sec);
  return waitFor(() => {
    expect(elements.button).toBeEnabled();
  })
    .then(() => {
      userEvent.click(elements.button);
    })
    .then(() => waitFor(() => {
      const text = getByText(document.body, 'Lorem ipsum 2020-07-02T10:46:16Z');
      expect(text).toBeInTheDocument();
      expect(getTree()).toMatchSnapshot();
    }))
    .then(() => {
      nock(proxyHost)
        .get(`/${rssFeedUrl2sec}`)
        .twice()
        .reply(200, rssXml2sec2);
    })
    .then(() => waitFor(() => {
      const text = getByText(document.body, 'Lorem ipsum 2020-07-02T10:46:20Z');
      expect(text).toBeInTheDocument();
      expect(getTree()).toMatchSnapshot();
    }, { timeout: 6000 }))
    .then(() => {
      nock(proxyHost)
        .get(`/${rssFeedUrl1sec}`)
        .twice()
        .reply(200, rssXml1sec1);
    })
    .then(() => {
      userEvent.paste(elements.input, rssFeedUrl1sec);
    })
    .then(() => waitFor(() => {
      expect(elements.button).toBeEnabled();
    }))
    .then(() => {
      userEvent.click(elements.button);
    })
    .then(() => waitFor(() => {
      const text = getByText(document.body, 'Lorem ipsum 2020-07-02T10:46:21Z');
      expect(text).toBeInTheDocument();
      expect(getTree()).toMatchSnapshot();
    }, { timeout: 6000 }));
}, 13000);
