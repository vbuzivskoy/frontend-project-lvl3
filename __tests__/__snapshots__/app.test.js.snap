// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`One rss feed test 1`] = `
"<div class=\\"jumbotron\\">
    <div class=\\"dropdown float-right\\">
        <button
            class=\\"btn btn-secondary btn-sm dropdown-toggle\\"
            type=\\"button\\"
            id=\\"localeDropdownMenuButton\\"
            data-toggle=\\"dropdown\\"
            aria-haspopup=\\"true\\"
            aria-expanded=\\"false\\"
        >
            en
        </button>
        <div
            id=\\"localeDropdownMenu\\"
            class=\\"dropdown-menu\\"
            aria-labelledby=\\"dropdownMenuButton\\"
        >
            <a class=\\"dropdown-item\\" href=\\"#\\">ru</a>
            <a class=\\"dropdown-item\\" href=\\"#\\">en</a>
        </div>
    </div>
    <h1 id=\\"appTitle\\">RSS Aggregator</h1>
    <form id=\\"rssFeedForm\\" class=\\"row\\">
        <div class=\\"col-sm-10\\">
            <div class=\\"form-group\\">
                <input
                    id=\\"inputField\\"
                    class=\\"form-control\\"
                    name=\\"feedUrl\\"
                    placeholder=\\"RSS link\\"
                />
                <div id=\\"validationFeedback\\" class=\\"invalid-feedback\\"></div>
            </div>
        </div>
        <div class=\\"col-sm-2\\">
            <button
                id=\\"submitButton\\"
                class=\\"btn btn-primary\\"
                type=\\"submit\\"
                disabled=\\"\\"
            >
                Add
            </button>
        </div>
    </form>
</div>
<div class=\\"container-fluid\\">
    <div class=\\"row\\">
        <div id=\\"rssChannels\\" class=\\"col-sm-5\\"></div>
        <div id=\\"rssPosts\\" class=\\"col-sm-7\\"></div>
    </div>
</div>
"
`;

exports[`One rss feed test 2`] = `
"<div class=\\"jumbotron\\">
    <div class=\\"dropdown float-right\\">
        <button
            class=\\"btn btn-secondary btn-sm dropdown-toggle\\"
            type=\\"button\\"
            id=\\"localeDropdownMenuButton\\"
            data-toggle=\\"dropdown\\"
            aria-haspopup=\\"true\\"
            aria-expanded=\\"false\\"
        >
            en
        </button>
        <div
            id=\\"localeDropdownMenu\\"
            class=\\"dropdown-menu\\"
            aria-labelledby=\\"dropdownMenuButton\\"
        >
            <a class=\\"dropdown-item\\" href=\\"#\\">ru</a>
            <a class=\\"dropdown-item\\" href=\\"#\\">en</a>
        </div>
    </div>
    <h1 id=\\"appTitle\\">RSS Aggregator</h1>
    <form id=\\"rssFeedForm\\" class=\\"row\\">
        <div class=\\"col-sm-10\\">
            <div class=\\"form-group\\">
                <input
                    id=\\"inputField\\"
                    class=\\"form-control\\"
                    name=\\"feedUrl\\"
                    placeholder=\\"RSS link\\"
                />
                <div id=\\"validationFeedback\\" class=\\"invalid-feedback\\"></div>
            </div>
        </div>
        <div class=\\"col-sm-2\\">
            <button
                id=\\"submitButton\\"
                class=\\"btn btn-primary\\"
                type=\\"submit\\"
                disabled=\\"\\"
            >
                Add
            </button>
        </div>
    </form>
</div>
<div class=\\"container-fluid\\">
    <div class=\\"row\\">
        <div id=\\"rssChannels\\" class=\\"col-sm-5\\">
            <div class=\\"card mb-2 bg-light border-0\\">
                <div class=\\"card-body\\">
                    <h5 class=\\"card-title\\">
                        Lorem ipsum feed for an interval of 2 seconds with 5
                        item(s)
                    </h5>
                    <p class=\\"card-text\\">
                        This is a constantly updating lorem ipsum feed.
                    </p>
                </div>
            </div>
        </div>
        <div id=\\"rssPosts\\" class=\\"col-sm-7\\">
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686776\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:16Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686774\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:14Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686772\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:12Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686770\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:10Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686768\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:08Z
                    </a>
                </div>
            </div>
        </div>
    </div>
</div>
"
`;

exports[`One rss feed test 3`] = `
"<div class=\\"jumbotron\\">
    <div class=\\"dropdown float-right\\">
        <button
            class=\\"btn btn-secondary btn-sm dropdown-toggle\\"
            type=\\"button\\"
            id=\\"localeDropdownMenuButton\\"
            data-toggle=\\"dropdown\\"
            aria-haspopup=\\"true\\"
            aria-expanded=\\"false\\"
        >
            en
        </button>
        <div
            id=\\"localeDropdownMenu\\"
            class=\\"dropdown-menu\\"
            aria-labelledby=\\"dropdownMenuButton\\"
        >
            <a class=\\"dropdown-item\\" href=\\"#\\">ru</a>
            <a class=\\"dropdown-item\\" href=\\"#\\">en</a>
        </div>
    </div>
    <h1 id=\\"appTitle\\">RSS Aggregator</h1>
    <form id=\\"rssFeedForm\\" class=\\"row\\">
        <div class=\\"col-sm-10\\">
            <div class=\\"form-group\\">
                <input
                    id=\\"inputField\\"
                    class=\\"form-control\\"
                    name=\\"feedUrl\\"
                    placeholder=\\"RSS link\\"
                />
                <div id=\\"validationFeedback\\" class=\\"invalid-feedback\\"></div>
            </div>
        </div>
        <div class=\\"col-sm-2\\">
            <button
                id=\\"submitButton\\"
                class=\\"btn btn-primary\\"
                type=\\"submit\\"
                disabled=\\"\\"
            >
                Add
            </button>
        </div>
    </form>
</div>
<div class=\\"container-fluid\\">
    <div class=\\"row\\">
        <div id=\\"rssChannels\\" class=\\"col-sm-5\\">
            <div class=\\"card mb-2 bg-light border-0\\">
                <div class=\\"card-body\\">
                    <h5 class=\\"card-title\\">
                        Lorem ipsum feed for an interval of 2 seconds with 5
                        item(s)
                    </h5>
                    <p class=\\"card-text\\">
                        This is a constantly updating lorem ipsum feed.
                    </p>
                </div>
            </div>
        </div>
        <div id=\\"rssPosts\\" class=\\"col-sm-7\\">
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686780\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:20Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686778\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:18Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686776\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:16Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686774\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:14Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686772\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:12Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686770\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:10Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686768\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:08Z
                    </a>
                </div>
            </div>
        </div>
    </div>
</div>
"
`;

exports[`Two rss feed test 1`] = `
"<div class=\\"jumbotron\\">
    <div class=\\"dropdown float-right\\">
        <button
            class=\\"btn btn-secondary btn-sm dropdown-toggle\\"
            type=\\"button\\"
            id=\\"localeDropdownMenuButton\\"
            data-toggle=\\"dropdown\\"
            aria-haspopup=\\"true\\"
            aria-expanded=\\"false\\"
        >
            en
        </button>
        <div
            id=\\"localeDropdownMenu\\"
            class=\\"dropdown-menu\\"
            aria-labelledby=\\"dropdownMenuButton\\"
        >
            <a class=\\"dropdown-item\\" href=\\"#\\">ru</a>
            <a class=\\"dropdown-item\\" href=\\"#\\">en</a>
        </div>
    </div>
    <h1 id=\\"appTitle\\">RSS Aggregator</h1>
    <form id=\\"rssFeedForm\\" class=\\"row\\">
        <div class=\\"col-sm-10\\">
            <div class=\\"form-group\\">
                <input
                    id=\\"inputField\\"
                    class=\\"form-control\\"
                    name=\\"feedUrl\\"
                    placeholder=\\"RSS link\\"
                />
                <div id=\\"validationFeedback\\" class=\\"invalid-feedback\\"></div>
            </div>
        </div>
        <div class=\\"col-sm-2\\">
            <button
                id=\\"submitButton\\"
                class=\\"btn btn-primary\\"
                type=\\"submit\\"
                disabled=\\"\\"
            >
                Add
            </button>
        </div>
    </form>
</div>
<div class=\\"container-fluid\\">
    <div class=\\"row\\">
        <div id=\\"rssChannels\\" class=\\"col-sm-5\\"></div>
        <div id=\\"rssPosts\\" class=\\"col-sm-7\\"></div>
    </div>
</div>
"
`;

exports[`Two rss feed test 2`] = `
"<div class=\\"jumbotron\\">
    <div class=\\"dropdown float-right\\">
        <button
            class=\\"btn btn-secondary btn-sm dropdown-toggle\\"
            type=\\"button\\"
            id=\\"localeDropdownMenuButton\\"
            data-toggle=\\"dropdown\\"
            aria-haspopup=\\"true\\"
            aria-expanded=\\"false\\"
        >
            en
        </button>
        <div
            id=\\"localeDropdownMenu\\"
            class=\\"dropdown-menu\\"
            aria-labelledby=\\"dropdownMenuButton\\"
        >
            <a class=\\"dropdown-item\\" href=\\"#\\">ru</a>
            <a class=\\"dropdown-item\\" href=\\"#\\">en</a>
        </div>
    </div>
    <h1 id=\\"appTitle\\">RSS Aggregator</h1>
    <form id=\\"rssFeedForm\\" class=\\"row\\">
        <div class=\\"col-sm-10\\">
            <div class=\\"form-group\\">
                <input
                    id=\\"inputField\\"
                    class=\\"form-control\\"
                    name=\\"feedUrl\\"
                    placeholder=\\"RSS link\\"
                />
                <div id=\\"validationFeedback\\" class=\\"invalid-feedback\\"></div>
            </div>
        </div>
        <div class=\\"col-sm-2\\">
            <button
                id=\\"submitButton\\"
                class=\\"btn btn-primary\\"
                type=\\"submit\\"
                disabled=\\"\\"
            >
                Add
            </button>
        </div>
    </form>
</div>
<div class=\\"container-fluid\\">
    <div class=\\"row\\">
        <div id=\\"rssChannels\\" class=\\"col-sm-5\\">
            <div class=\\"card mb-2 bg-light border-0\\">
                <div class=\\"card-body\\">
                    <h5 class=\\"card-title\\">
                        Lorem ipsum feed for an interval of 2 seconds with 5
                        item(s)
                    </h5>
                    <p class=\\"card-text\\">
                        This is a constantly updating lorem ipsum feed.
                    </p>
                </div>
            </div>
        </div>
        <div id=\\"rssPosts\\" class=\\"col-sm-7\\">
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686776\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:16Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686774\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:14Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686772\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:12Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686770\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:10Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686768\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:08Z
                    </a>
                </div>
            </div>
        </div>
    </div>
</div>
"
`;

exports[`Two rss feed test 3`] = `
"<div class=\\"jumbotron\\">
    <div class=\\"dropdown float-right\\">
        <button
            class=\\"btn btn-secondary btn-sm dropdown-toggle\\"
            type=\\"button\\"
            id=\\"localeDropdownMenuButton\\"
            data-toggle=\\"dropdown\\"
            aria-haspopup=\\"true\\"
            aria-expanded=\\"false\\"
        >
            en
        </button>
        <div
            id=\\"localeDropdownMenu\\"
            class=\\"dropdown-menu\\"
            aria-labelledby=\\"dropdownMenuButton\\"
        >
            <a class=\\"dropdown-item\\" href=\\"#\\">ru</a>
            <a class=\\"dropdown-item\\" href=\\"#\\">en</a>
        </div>
    </div>
    <h1 id=\\"appTitle\\">RSS Aggregator</h1>
    <form id=\\"rssFeedForm\\" class=\\"row\\">
        <div class=\\"col-sm-10\\">
            <div class=\\"form-group\\">
                <input
                    id=\\"inputField\\"
                    class=\\"form-control\\"
                    name=\\"feedUrl\\"
                    placeholder=\\"RSS link\\"
                />
                <div id=\\"validationFeedback\\" class=\\"invalid-feedback\\"></div>
            </div>
        </div>
        <div class=\\"col-sm-2\\">
            <button
                id=\\"submitButton\\"
                class=\\"btn btn-primary\\"
                type=\\"submit\\"
                disabled=\\"\\"
            >
                Add
            </button>
        </div>
    </form>
</div>
<div class=\\"container-fluid\\">
    <div class=\\"row\\">
        <div id=\\"rssChannels\\" class=\\"col-sm-5\\">
            <div class=\\"card mb-2 bg-light border-0\\">
                <div class=\\"card-body\\">
                    <h5 class=\\"card-title\\">
                        Lorem ipsum feed for an interval of 2 seconds with 5
                        item(s)
                    </h5>
                    <p class=\\"card-text\\">
                        This is a constantly updating lorem ipsum feed.
                    </p>
                </div>
            </div>
        </div>
        <div id=\\"rssPosts\\" class=\\"col-sm-7\\">
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686780\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:20Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686778\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:18Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686776\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:16Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686774\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:14Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686772\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:12Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686770\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:10Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686768\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:08Z
                    </a>
                </div>
            </div>
        </div>
    </div>
</div>
"
`;

exports[`Two rss feed test 4`] = `
"<div class=\\"jumbotron\\">
    <div class=\\"dropdown float-right\\">
        <button
            class=\\"btn btn-secondary btn-sm dropdown-toggle\\"
            type=\\"button\\"
            id=\\"localeDropdownMenuButton\\"
            data-toggle=\\"dropdown\\"
            aria-haspopup=\\"true\\"
            aria-expanded=\\"false\\"
        >
            en
        </button>
        <div
            id=\\"localeDropdownMenu\\"
            class=\\"dropdown-menu\\"
            aria-labelledby=\\"dropdownMenuButton\\"
        >
            <a class=\\"dropdown-item\\" href=\\"#\\">ru</a>
            <a class=\\"dropdown-item\\" href=\\"#\\">en</a>
        </div>
    </div>
    <h1 id=\\"appTitle\\">RSS Aggregator</h1>
    <form id=\\"rssFeedForm\\" class=\\"row\\">
        <div class=\\"col-sm-10\\">
            <div class=\\"form-group\\">
                <input
                    id=\\"inputField\\"
                    class=\\"form-control\\"
                    name=\\"feedUrl\\"
                    placeholder=\\"RSS link\\"
                />
                <div id=\\"validationFeedback\\" class=\\"invalid-feedback\\"></div>
            </div>
        </div>
        <div class=\\"col-sm-2\\">
            <button
                id=\\"submitButton\\"
                class=\\"btn btn-primary\\"
                type=\\"submit\\"
                disabled=\\"\\"
            >
                Add
            </button>
        </div>
    </form>
</div>
<div class=\\"container-fluid\\">
    <div class=\\"row\\">
        <div id=\\"rssChannels\\" class=\\"col-sm-5\\">
            <div class=\\"card mb-2 bg-light border-0\\">
                <div class=\\"card-body\\">
                    <h5 class=\\"card-title\\">
                        Lorem ipsum feed for an interval of 1 seconds with 5
                        item(s)
                    </h5>
                    <p class=\\"card-text\\">
                        This is a constantly updating lorem ipsum feed.
                    </p>
                </div>
            </div>
            <div class=\\"card mb-2 bg-light border-0\\">
                <div class=\\"card-body\\">
                    <h5 class=\\"card-title\\">
                        Lorem ipsum feed for an interval of 2 seconds with 5
                        item(s)
                    </h5>
                    <p class=\\"card-text\\">
                        This is a constantly updating lorem ipsum feed.
                    </p>
                </div>
            </div>
        </div>
        <div id=\\"rssPosts\\" class=\\"col-sm-7\\">
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686781\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:21Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686779\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:19Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686777\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:17Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686780\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:20Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686778\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:18Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686776\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:16Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686774\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:14Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686772\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:12Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686770\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:10Z
                    </a>
                </div>
            </div>
            <div class=\\"card mb-1 bg-light border-0\\">
                <div class=\\"card-body p-1 pl-3\\">
                    <a
                        class=\\"card-text\\"
                        href=\\"http://example.com/test/1593686768\\"
                        target=\\"_blank\\"
                    >
                        Lorem ipsum 2020-07-02T10:46:08Z
                    </a>
                </div>
            </div>
        </div>
    </div>
</div>
"
`;
